pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -507,8 +432,240 @@ mv $(workspaces.source.path)/source/.jbs/build.sh $(workspaces.source.path)`, or
+		Params:     append(pipelineParams, tektonpipeline.ParamSpec{Name: PipelineResultPreBuildImageDigest, Type: tektonpipeline.ParamTypeString}),
+		Results: []tektonpipeline.TaskResult{
+			{Name: PipelineResultContaminants},
+			{Name: PipelineResultDeployedResources},
+			{Name: PipelineResultPassedVerification},
+			{Name: PipelineResultVerificationResult},
+			// TODO: ### DeployPreBuildSource and Deploy push to git. Currently the former is used for GitArchive results.
+			//			{Name: PipelineResultGitArchive},
+		},
+		Steps: []tektonpipeline.Step{
+			{
+				Name:            "restore-post-build-artifacts",
+				Image:           strings.TrimSpace(strings.Split(buildTrustedArtifacts, "FROM")[1]),
+				ImagePullPolicy: v1.PullIfNotPresent,
+				SecurityContext: &v1.SecurityContext{RunAsUser: &zero},
+				Env:             secretVariables,
+				// While the manifest digest is available we need the manifest of the layer within the archive hence
+				// using 'oras manifest fetch' to extract the correct layer.
+				Script: fmt.Sprintf(`echo "Restoring artifacts to workspace : $(workspaces.source.path)"
+export ORAS_OPTIONS="%s"
+URL=%s
+DIGEST=$(tasks.%s.results.IMAGE_DIGEST)
+AARCHIVE=$(oras manifest fetch $ORAS_OPTIONS $URL@$DIGEST | jq --raw-output '.layers[0].digest')
+echo "URL $URL DIGEST $DIGEST AARCHIVE $AARCHIVE"
+use-archive oci:$URL@$AARCHIVE=$(workspaces.source.path)/artifacts`, orasOptions, registryArgsWithDefaults(jbsConfig, ""), BuildTaskName),
+			},
+			{
+				Name:            "verify-and-check-for-contaminates",
+				Image:           buildRequestProcessorImage,
+				ImagePullPolicy: pullPolicy,
+				SecurityContext: &v1.SecurityContext{RunAsUser: &zero},
+				Env:             secretVariables,
+				ComputeResources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{"memory": limits.defaultBuildRequestMemory, "cpu": limits.defaultRequestCPU},
+					Limits:   v1.ResourceList{"memory": limits.defaultBuildRequestMemory, "cpu": limits.defaultLimitCPU},
+				},
+				Script: buildTaskScript,
+			},
+			// TODO: ### Do we really need this now. Previously it stored:
+			// 		artifacts (which is now handled above)
+			//		source - useful to store pre-post-build script changes?
+			//		logs - irrelevant if build is done in container?
+			//{
+			//	Name:            "create-post-build-image",
+			//	Image:           strings.TrimSpace(strings.Split(buildTrustedArtifacts, "FROM")[1]),
+			//	ImagePullPolicy: v1.PullIfNotPresent,
+			//	SecurityContext: &v1.SecurityContext{RunAsUser: &zero},
+			//	Env:             secretVariables,
+			//	ComputeResources: v1.ResourceRequirements{
+			//		Requests: v1.ResourceList{"memory": limits.defaultBuildRequestMemory, "cpu": limits.defaultRequestCPU},
+			//		Limits:   v1.ResourceList{"memory": limits.defaultBuildRequestMemory, "cpu": limits.defaultLimitCPU},
+			//	},
+			//	Script: postBuildImageArgs,
```

Comment:
@vibe13 to be discussed - I don't think we need this anymore.

Reply:
Ok to remove the storage of the artifacts which are already stored with "store-post-build-artifacts" task. I would store any change to the scripts, for a full attestation. Wrt the logs, not sure I completely follow, why is it irrelevant if done in a container? I would still store them (regardless if it's a container build or a normal build). Additionally we will store logs in PNC DB. If later on we feel it's not necessary anymore, we can remove the storage of logs here.